### PR TITLE
Refactored ProcessingParameters. Also fixes bug with parameters on LSF.

### PR DESCRIPTION
**Pull request by vinjana**

### CODE REVIEW

File: src/main/groovy/de/dkfz/roddy/execution/jobs/cluster/lsf/LSFCommand.groovy [hunk 1]
```diff
@@ -233,7 +217,7 @@ class LSFCommand extends Command {
     private String prepareParentJobs(List<BEJobID> jobIds) {
         List<BEJobID> validJobIds = BEJob.uniqueValidJobIDs(jobIds)
         if (validJobIds.size() > 0) {
-            String joinedParentJobs = validJobIds.collect { "done(${it})" }.join(" && ")
+            String joinedParentJobs = validJobIds.collect { "exit(${it}, 0)" }.join(" && ")
```
Comment (dankwart-de) [inline src/main/groovy/de/dkfz/roddy/execution/jobs/cluster/lsf/LSFCommand.groovy]: What I initially intended was, that you can control job dependencies using processingcommands. Never really was used but we might want to support this mechanism in some other way in the future.

File: src/main/groovy/de/dkfz/roddy/execution/jobs/BatchEuphoriaJobManager.groovy [hunk 2]
```diff
@@ -109,10 +110,10 @@ abstract class BatchEuphoriaJobManager<C extends Command> {
         })
     }
 
-    abstract C createCommand(BEJob job, String jobName, List<ProcessingCommands> processingCommands, File tool, Map<String, String> parameters, List<String> parentJobs)
+    abstract C createCommand(BEJob job, String jobName, List<ProcessingParameters> processingParameters, File tool, Map<String, String> parameters, List<String> parentJobs)
 
     C createCommand(BEJob job, File tool, List<String> parentJobs, Map<String, String> parameters) {
-        C c = (C) createCommand(job, job.jobName, job.getListOfProcessingCommand(), tool, parameters, parentJobs)
+        C c = (C) createCommand(job, job.jobName, job.getListOfProcessingParameters(), tool, parameters, parentJobs)
```
[the remark this answers is not in the document]
Comment (vinjana) [reply src/main/groovy/de/dkfz/roddy/execution/jobs/BatchEuphoriaJobManager.groovy]: Yes, that is more specific. Maybe still we may want to call it "resourceParameters". Apparently it is somehow possible to add more other types of "ProcessingParameter", but the responsible function (`de.dkfz.roddy.execution.jobs.BEJob#addProcessingParameters`) seems to be unused.

File: src/main/groovy/de/dkfz/roddy/execution/jobs/cluster/lsf/LSFCommand.groovy (hunk 1 above)
Comment (vinjana) [reply src/main/groovy/de/dkfz/roddy/execution/jobs/cluster/lsf/LSFCommand.groovy]: But what are ProcessingCommands?

File: src/main/groovy/de/dkfz/roddy/execution/jobs/BatchEuphoriaJobManager.groovy (hunk 2 above)
[the remark this answers is not in the document]
Comment (dankwart-de) [reply src/main/groovy/de/dkfz/roddy/execution/jobs/BatchEuphoriaJobManager.groovy]: It is currently unused. It was at some point, where you could change afterok to after or override the resource set manually by setting up a configuration value for the tool.

File: src/main/groovy/de/dkfz/roddy/execution/jobs/cluster/lsf/LSFCommand.groovy (hunk 1 above)
Comment (dankwart-de) [reply src/main/groovy/de/dkfz/roddy/execution/jobs/cluster/lsf/LSFCommand.groovy]: As said above, they were used to control the submission behaviour of Roddy jobs. But as also mentionend above, the name never was good. So SubmissionDirectives would be much more descriptive.

**Pull request merged**